### PR TITLE
Jackson/quick build dev

### PR DESCRIPTION
**Pull request by jvstokes**
flexible source location, vcs ignore support, flexible dockerfile path.

### CODE REVIEW

File: commands/azureCommands/acr-build.ts [hunk 1]
```diff
@@ -112,3 +107,12 @@ function getTempSourceArchivePath(): string {
     let tarFilePath = url.resolve(os.tmpdir(), `sourceArchive${id}.tar.gz`);
     return tarFilePath;
 }
+
+function filter(list: string[]): string[] {
+    for (let file of list) {
+        if (vcsIgnoreList.indexOf(file) !== -1) {
+            list.splice(list.indexOf(file), 1);
```
Comment (northtyphoon) [inline commands/azureCommands/acr-build.ts]: I'm always cautious about updating the list in the enumeration. Can you hold the filtered item in a new list and return the new list?

File: commands/azureCommands/acr-build.ts [hunk 2]
```diff
@@ -55,35 +51,39 @@ export async function queueBuild(dockerFileUri?: vscode.Uri): Promise<void> {
...
+async function uploadSourceCode(client: ContainerRegistryManagementClient, registryName: string, resourceGroupName: string, sourceLocation: string, tarFilePath: string, folder: vscode.WorkspaceFolder): Promise<string> {
     status.appendLine("   Sending source code to temp file");
-    tar.c(
-        {
-            gzip: true
-        },
-        [sourceLocation]
-    ).pipe(fs.createWriteStream(tarFilePath));
+    let source = sourceLocation.substring(1);
+    let current = process.cwd();
```
Comment (northtyphoon) [inline commands/azureCommands/acr-build.ts]: The code doesn't restore the current folder

Comment (northtyphoon) [reply commands/azureCommands/acr-build.ts]: If you restore the current dir before tar, how does tar.c know the correct item path? The behavior is very unusual if it works like you describe

Comment (jvstokes) [reply commands/azureCommands/acr-build.ts]: Oh you're right, forgot to check tar. Fixed!

**Pull request merged**